scripts/Dockerfile.build

```diff
@@ -17,10 +18,17 @@ WORKDIR /tmp/dd/datadog-agent/cmd/serverless
 # add the current version number to the tags package before compilation
 
 RUN --mount=type=cache,target=/go/pkg/mod \
-  --mount=type=cache,target=/root/.cache/go-build \
-  go build -ldflags="-w -s \
-  -X github.com/DataDog/datadog-agent/pkg/serverless/tags.currentExtensionVersion=$EXTENSION_VERSION" \
-  -tags serverless -o datadog-agent
+    --mount=type=cache,target=/root/.cache/go-build \ 
+    if [ -z "$AGENT_VERSION" ]; then \
+        go build -ldflags="-w -s \
+        -X github.com/DataDog/datadog-agent/pkg/serverless/tags.currentExtensionVersion=$EXTENSION_VERSION" \
+        -tags serverless -o datadog-agent; \
+    else \
+        go build -ldflags="-w -s \
+        -X github.com/DataDog/datadog-agent/pkg/serverless/tags.currentExtensionVersion=$EXTENSION_VERSION \
+        -X github.com/DataDog/datadog-agent/pkg/version.agentVersionDefault=$AGENT_VERSION" \
```

Comment:
am I.bit afraid that, since this variable is not in serverless owned files, this could change and break our build, potentially silently?
what happen if the variable is renamed or deleted? does it fail explicitly or silently?

Reply:
if the variable is renamed, our agent version override fails and the extension falls back on using the default agent version (which we are currently using) in the metrics intake API calls. Perhaps I can do a check that the variable exists first?

Reply:
is there a way of doing this with `go` cli?

Reply:
I'm investigating with `go tool nm`: https://pkg.go.dev/cmd/nm

Reply:
added a check to fail the script if the variable doesnt exist (or is renamed)